### PR TITLE
Add buildah dockerfiles

### PR DESCRIPTION
**Pull request by chmouel**
# Changes

Add DockerFile for Buildah task

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ X ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ X ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: buildah/Dockerfiles/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM centos:7
```
Comment (siamaksade) [inline buildah/Dockerfiles/Dockerfile]: Should be based on UBI 8

Comment (chmouel) [reply buildah/Dockerfiles/Dockerfile]: i can't find buildah and podman package on ubi 8  :(

File: buildah/Dockerfiles/Makefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+REGISTRY=quay.io/openshift-pipeline/buildah
```
Comment (dlorenc) [inline buildah/Dockerfiles/Makefile]: We won't have permission to push here right? Maybe we should figure out a home for all the images in the catalog.

Comment (chmouel) [reply buildah/Dockerfiles/Makefile]: @vdemeester what do you think ?

Comment (vdemeester) [reply buildah/Dockerfiles/Makefile]: agreeing with @dlorenc, in the future we may want to have some common place where the image defined here in the catalog are pushed.

**Comment by dlorenc** [comment]
Ref https://github.com/tektoncd/catalog/issues/29

**Comment by vbatts** [comment]
is the goal of the catalog to have yaml, and possibly Dockerfiles, ready for folks to use?
I feel this PR is not quite re-using the Dockerfile that upstream buildah provides, nor the fresh builds of it that live at quay.io/openshift-pipeline/buildah.
I feel that it might only need to be the yaml to make these things available in tekton as resources.

**Comment by dlorenc** [comment]
> I feel that it might only need to be the yaml to make these things available in tekton as resources.

Yeah I think we want to prefer to use existing images, where possible. If for some reason Tekton requires a differently configured image we could consider hosting the Dockerfile here, but I don't think that should be a common case.

**Comment by chmouel** [comment]
The idea at first was to have the s2i binary as well, altought we don't need to have an image containing all so let's use `quay.io/openshift-pipeline/buildah`

**Pull request closed without merging**